### PR TITLE
[node] Repro concurrency issues that Connext has been seeing

### PR DESCRIPTION
**Pull request by rhlsthrm**
To repro, run the following test in `packages/node`:

`yarn test connext-issue.spec.ts`

### CODE REVIEW

**Comment by snario** [comment]
Would recommend cleaning up the files a bit to put helpers for the connext apps in an isolated place

File: packages/node/src/message-handling/handle-protocol-message.ts [hunk 1]
```diff
@@ -234,6 +232,14 @@ async function getQueueNamesListByProtocolName(
 
       return [appIdentityHash];
 
+    case Protocol.Uninstall:
```
Comment (LayneHaber) [inline packages/node/src/message-handling/handle-protocol-message.ts]: 🔥

**Comment by LayneHaber** [comment]
> Would recommend cleaning up the files a bit to put helpers for the connext apps in an isolated place

can we assign this to me and put it in a new issue? dont want to have this block merging the PR (but happy to do it once we've verified the flow)

File: packages/apps/test/linked-transfer-app.spec.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
   solidityKeccak256
 } from "ethers/utils";
 
-import UnidirectionalLinkedTransferApp from "../expected-build-artifacts/UnidirectionalLinkedTransferApp.json";
+import UnidirectionalLinkedTransferApp from "../build/UnidirectionalLinkedTransferApp.json";
```
Comment (snario) [inline packages/apps/test/linked-transfer-app.spec.ts]: Looks wrong

File: packages/node/src/methods/app-instance/install-virtual/controller.ts [hunk 3]
```diff
@@ -21,27 +21,26 @@ export default class InstallVirtualController extends NodeController {
...
+      [publicIdentifier, proposedToIdentifier],
+      networkContext.ProxyFactory,
+      networkContext.MinimumViableMultisig
+    );
 
-    return queues;
+    return [
+      requestHandler.getShardedQueue(multisigAddressWithHub),
+      requestHandler.getShardedQueue(multisigAddressWithResponding)
+    ];
```
Comment (snario) [inline packages/node/src/methods/app-instance/install-virtual/controller.ts]: Perhaps split this into a separate PR @cf19drofxots

File: packages/node/src/methods/app-instance/uninstall/controller.ts [hunk 4]
```diff
@@ -25,11 +25,15 @@ export default class UninstallController extends NodeController {
...
+
     if (sc.freeBalance.identityHash === appInstanceId) {
       throw Error(CANNOT_UNINSTALL_FREE_BALANCE(sc.multisigAddress));
     }
 
-    return [requestHandler.getShardedQueue(sc.multisigAddress)];
+    return [
+      requestHandler.getShardedQueue(sc.multisigAddress),
+      requestHandler.getShardedQueue(appInstanceId)
+    ];
```
Comment (snario) [inline packages/node/src/methods/app-instance/uninstall/controller.ts]: Also probably separate

File: packages/node/test/integration/unidirectional-transfer.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { AppABIEncodings } from "@counterfactual/types";
```
Comment (snario) [inline packages/node/test/integration/unidirectional-transfer.ts]: This and some other files are helpers / utils for a specific set of contracts and should not be in the `test/integration/*` path, maybe put in a subfolder
